source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -347,13 +354,27 @@ bool SurfaceManipulationWidget::onMouseUp_( Viewer::MouseButton button, int /*mo
                         .multipleEdgesResolveMode = FillHoleParams::MultipleEdgesResolveMode::Strong
                     },
                     .maxEdgeLen = 2 * (float)avgLen,
+                    .beforeEdgeSplit = [&] ( EdgeId e ) -> bool
+                    {
+                        if ( contains( unchangeableEdges, e.undirected() ) )
+                            return false;
+                        return true;
+                    },
```

Comment:
I believe that we should not use this,  it might decrease quality of the result

Reply:
reworked